src/Composer/ComposerWidget.vala

```diff
@@ -77,11 +78,29 @@ public class Mail.ComposerWidget : Gtk.Grid {
         Object (has_subject_field: true);
     }
 
+    public ComposerWidget.from_request ( ComposeRequest req ) {
+        Object (has_subject_field: true, template: req);
+    }
+
     construct {
         actions = new SimpleActionGroup ();
         actions.add_action_entries (action_entries, this);
         insert_action_group (ACTION_GROUP_PREFIX, actions);
 
+        var recipient_grid = construct_header_fields ();
```

Comment:
Note that I've done a refactor of the `construct` block here. I've re-arranged the code into four different methods, each of which returns its top-level element, more or less. This does make it a little hard to find the mailto related changes, but if you grep for `template` you'll see them.

Reply:
I would really like to avoid using methods for this. It doesn't feel very object-oriented to me. If it feels cleaner to not have these widgets constructed here then we should probably create classes instead of methods

Reply:
I do believe that the existing constructor method is way too long for a single function, and adding classes is overcomplicating things. Functional deconstruction seems the cleanest approach. That said, I noticed you have a pending PR that this would create a messy merge conflict with, so best to save this refactor for later.